src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -131,37 +135,51 @@ public static Version GetInstalledAsmVersion2(IEnumerable<Version> versions, ref
                 getASMInstallsFunc = getASMInstallsFunc ?? GetAsmInstallations;
                 var installations = getASMInstallsFunc(rootFolder);
 
-                //first find the closest matches using major, minor and build version.
+                // first find the exact match or the lowest matching within same major version
                 foreach (var version in versions)
                 {
+                    Dictionary<Version, string> versionToLocationDic = new Dictionary<Version, string>();
                     foreach (KeyValuePair<string, Tuple<int, int, int, int>> install in installations)
                     {
                         var installVersion = new Version(install.Value.Item1, install.Value.Item2, install.Value.Item3);
-                        if (version.Major == installVersion.Major &&
-                            version.Minor == installVersion.Minor &&
-                            version.Build == installVersion.Build)
+                        if (version.Major == installVersion.Major)
+                        {
+                            versionToLocationDic.Add(installVersion, install.Key);
+                        }
+                    }
+
+                    // When there is major version matching, continue the search
+                    if (versionToLocationDic.Count != 0)
+                    {
+                        versionToLocationDic.TryGetValue(version, out location);
+                        // If exact matching version found, return it
+                        if (location != null)
                         {
-                            location = install.Key;
                             return version;
                         }
+                        // If no matching version, return the lowest within same major
+                        else
+                        {
+                            location = versionToLocationDic[versionToLocationDic.Keys.Min()];
+                            return versionToLocationDic.Keys.Min();
+                        }
                     }
                 }
 
-                //Fallback mechanism, look inside libg folders if any of them
-                //contain ASM dlls.
+                // Fallback mechanism, look inside libg folders if any of them contains ASM dlls.
                 foreach (var v in versions)
```

Comment:
I wonder if we should have this fallback happen first in order to make debugging and development easier...

Reply:
I understand where you started but I think this new order could be confusing for larger audience and 3rd party developers, if we want ASM from libG folder to be loaded, we can clear the property `ProductsWithASM`, what do you think?